vignettes/building_mocks.Rmd

```diff
@@ -0,0 +1,533 @@
+---
+title: "Building Mocks"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Building Mocks}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r message=FALSE}
+library(tfrmt)
+library(dplyr) # for arrange_all()
+library(tidyr) # for crossing()
+```
+
+Mock tables are useful for visualising how future generated data may be 
+presented in its final form. Using the `{tfrmt}` package you can create a mock 
+table using placeholders (e.g. `xx.x`, `XXX`) or using mock data with numerical 
+values. Note that using the lower or upper case is up to your preference.
+
+Tables are generally built by creating a tfrmt specification tailored to your 
+data, and then feeding that into one of the available print functions. In this 
+article we will only explore the use of the `print_mock_gt()` function. For 
+information on `print_to_gt()` please refer to its documentation.
+
+All tfrmt specifications are created using the `tfrmt()` function. The 
+foundation of this function are the `label`, `column`, `param` and `values` 
+parameters. From here you can specify a `body_plan` where custom 
+`frmt_structures` can be added and built upon.
+
+**Note:** layering has been used to avoid repeating code; for 
```

Comment:
If possible, I think we should avoid the use of building template functions in this vignette. That is a more advanced topic while mocks are not.